src/core/models/answersconfig.js

```diff
@@ -0,0 +1,216 @@
+/** @module AnswersConfig */
+
+import { LOCALE, PRODUCTION, QUERY_SOURCE, SANDBOX } from '../constants';
+import SearchConfig from '../models/searchconfig';
+import VerticalPagesConfig from '../models/verticalpagesconfig';
+
+const sandboxPrefix = `${SANDBOX}-`;
+const DEFAULTS = {
+  environment: PRODUCTION,
+  onReady: function () {},
+  useTemplates: true,
+  locale: LOCALE,
+  experienceVersion: 'PRODUCTION',
+  debug: false,
+  sessionTrackingEnabled: true,
+  onStateChange: function () {},
+  onVerticalSearch: function () {},
+  onUniversalSearch: function () {},
+  disableCssVariablesPonyfill: false,
+  querySource: QUERY_SOURCE
+};
+
+export default class AnswersConfig {
```

Comment:
What would you think about having two classes: `AnswersConfig` and an `AnswersConfigBuilder`. The role of the latter would be to create the former. The validation logic would live in the `AnswersConfigBuilder#build` method. Here's an article about the builder pattern in JS: https://medium.com/better-programming/the-builder-pattern-in-javascript-6f3d85c3ae4a. It might be nice to separate out the actual `AnswersConfig` data model from how it is constructed. The call chain could be something like:

```
const answersConfig = new AnswersConfigBuilder()
.setRawConfig(config)
.build();
```

The `AnswersConfig` would basically be a wrapper around the object it's passed, nothing too fancy. In the future, we might do something more interesting with the `AnswersConfig` model. If we ever use TypeSciprt, we can make it a strongly typed concept.

Reply:
I second this idea. Something else to keep in mind is that the validation is currently happening on the parsed config after the defaults have been applied. I think it makes more sense to perform the validation on the raw config, and Tom's suggested changes might make that easier to do